frontend/src/workStation/CodeGroup.jsx

```diff
@@ -0,0 +1,352 @@
+  HStack,
+  useDisclosure,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+} from "@chakra-ui/react";
+
+function CodeGroup() {
+  const [groupData, setGroupdata] = useState([]);
+  const [allFiles, setAllFiles] = useState([]);
+  const [allMembers, setAllMembers] = useState([]);
+  const [newGroupName, setNewGroupName] = useState("");
+  const [newGroupImageUrl, setNewGroupImageUrl] = useState("");
+  const [newMemberInput, setNewMemberInput] = useState("");
+  const [newMembers, setNewMembers] = useState([]);
+  const [newFileName, setNewFileName] = useState("");
+  const [error, setError] = useState("");
+  const {
+    isOpen: isUpdateGroupModalOpen,
+    onOpen: onUpdateGroupModalOpen,
+    onClose: onUpdateGroupModalClose,
+  } = useDisclosure();
+  const {
+    isOpen: isAddMembersModalOpen,
+    onOpen: onAddMembersModalOpen,
+    onClose: onAddMembersModalClose,
+  } = useDisclosure();
+  const {
+    isOpen: isCreateNewFileModalOpen,
+    onOpen: onCreateNewFileModalOpen,
+    onClose: onCreateNewFileModalClose,
+  } = useDisclosure();
+
+  const { groupId } = useParams();
+  const BACKEND_URL = import.meta.env.VITE_BACKEND_URL;
+  const token = localStorage.getItem("jwt");
+  const navigate = useNavigate();
+  const options = {
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+    withCredentials: true,
+  };
+
+  const fetchGroupDetails = async () => {
+    try {
+      const response = await axios.get(
+        `${BACKEND_URL}/api/group/${groupId}`,
+        options
+      );
+      setGroupdata(response.data);
+      setAllFiles(response.data.files);
+      setAllMembers(response.data.members);
+    } catch (error) {
+      setError("Error fetching group details");
+    }
+  };
+
+  useEffect(() => {
+    fetchGroupDetails();
+  }, []);
+
+  const handleUpdateGroupDetails = async () => {
+    try {
+      const response = await axios.put(
+        `${BACKEND_URL}/api/group/${groupId}/update-code-group`,
+        { newGroupName, newGroupImageUrl, newMembers },
+        options
+      );
+    } catch (error) {
+      setError("Error updating group details");
+    } finally {
+      setNewGroupName("");
+      setNewGroupImageUrl("");
+      setNewMembers([]);
+      fetchGroupDetails();
+      onUpdateGroupModalClose();
+      onAddMembersModalClose();
+    }
+  };
+
+  function handleAddNewMemberToList() {
+    if (newMemberInput) {
+      setNewMembers([...newMembers, newMemberInput]);
+      setNewMemberInput("");
+    }
+  }
+
+  function handleRemoveNewMemberToList(memberUsername) {
+    setNewMembers(newMembers.filter((member) => member !== memberUsername));
+  }
+
+  const handleCreateNewFile = async () => {
+    try {
+      const response = await axios.post(
+        `${BACKEND_URL}/group/${groupId}/create-new-file`,
+        { newFileName },
+        options
+      );
+    } catch (error) {
+      setError("Error Creating Group. Try again");
```

Comment:
This looks like the wrong error message

Reply:
Thanks. Fixed here: https://github.com/izuchukwumba/peered/commit/a960d993a9510bf0b293543fdb359403c740e67f